pkg/kyverno/common/common.go

```diff
@@ -947,16 +951,27 @@ func PrintMutatedPolicy(mutatedPolicies []*v1.ClusterPolicy) error {
 	return nil
 }
 
-func CheckVariableForPolicy(valuesMap map[string]map[string]Resource, policyName string, resourceName string, resourceKind string, variables map[string]string, kindOnwhichPolicyIsApplied map[string]struct{}, variable string) (map[string]string, error) {
+func CheckVariableForPolicy(valuesMap map[string]map[string]Resource, globalValMap map[string]string, policyName string, resourceName string, resourceKind string, variables map[string]string, kindOnwhichPolicyIsApplied map[string]struct{}, variable string) (map[string]string, error) {
 	// get values from file for this policy resource combination
 	thisPolicyResourceValues := make(map[string]string)
 	if len(valuesMap[policyName]) != 0 && !reflect.DeepEqual(valuesMap[policyName][resourceName], Resource{}) {
 		thisPolicyResourceValues = valuesMap[policyName][resourceName].Values
 	}
+
 	for k, v := range variables {
 		thisPolicyResourceValues[k] = v
 	}
 
+	if thisPolicyResourceValues == nil && len(globalValMap) > 0 {
+		thisPolicyResourceValues = make(map[string]string)
```

Comment:
why do we need to initialise `thisPolicyResourceValues` again? Any specific reason?

Reply:
To handle this edge case:
```yaml
policies:
  - name: cm-globalval-example
    resources:
      - name: test-global-dev
      - name: test-global-prod
        values:
          request.mode: prod
globalValues:
  request.mode: dev
```
We can eliminate it otherwise.